tests/scripts/thread-cert/Cert_5_2_06_RouterDowngrade.py

```diff
@@ -31,43 +31,72 @@
 import unittest
 
 import node
+import config
+import command
+import mle
 
 LEADER = 1
-ROUTER1 = 2
+DUT_ROUTER1 = 2
+ROUTER2 = 3
+ROUTER24 = 24
 
 class Cert_5_2_06_RouterDowngrade(unittest.TestCase):
     def setUp(self):
         self.nodes = {}
-        for i in range(1, 26):
+        for i in range(1, 25):
             self.nodes[i] = node.Node(i)
-            self.nodes[i].set_panid(0xface)
+            self.nodes[i].set_panid()
             self.nodes[i].set_mode('rsdn')
             self.nodes[i].set_router_selection_jitter(1)
-            if i != ROUTER1:
+            if i != DUT_ROUTER1:
                 self.nodes[i].set_router_upgrade_threshold(32)
                 self.nodes[i].set_router_downgrade_threshold(32)
 
+        self.sniffer = config.create_default_thread_sniffer()
+        self.sniffer.start()
+
     def tearDown(self):
+        self.sniffer.stop()
+        del self.sniffer
+
         for node in list(self.nodes.values()):
             node.stop()
         del self.nodes
 
     def test(self):
+        # 1 Ensure topology is formed correctly without the ROUTER24.
         self.nodes[LEADER].start()
         self.nodes[LEADER].set_state('leader')
         self.assertEqual(self.nodes[LEADER].get_state(), 'leader')
 
-        for i in range(2, 25):
+        for i in range(2, 24):
             self.nodes[i].start()
             time.sleep(5)
             self.assertEqual(self.nodes[i].get_state(), 'router')
 
-        self.nodes[25].start()
+        # This method flushes the message queue so calling this method again will return only the newly logged messages.
+        dut_messages = self.sniffer.get_messages_sent_by(DUT_ROUTER1)
+
+        # 2 ROUTER24: Attach to network. We don't need to ensure it has a better link of quality on posix.
```

Comment:
Why is posix special?

nit: `link of quality` -> `link quality`

Reply:
Quoting Step 2 of Test Plan here:
> Harness causes Router_24 to attach to the network and ensures it has a link of quality 2 or better to Router_1 and Router_2

No special whitelist or blacklist setting in this case. I think the intention of such requirement is to consider in real test scenarios Router24 may have a poor link connection. However in posix, the link quality should not be an issue. Besides, all reference testbed devices have been configured with downgrade threshold as 32 except the DUT, so only DUT has the chance to downgrade even without setting the link quality.

Thoughts?